### ISSUE TITLE
Running the opendatacon?

### ISSUE THREAD

**Issue by rrivera1849** [neilstephens/opendatacon#9]
I see that there are several scripts in the install directory to run the opendatacon daemon.

When I run the init script I get the following error:
/etc/init.d/functions: No such file or directory

My assumption is that I have to do something else other than building and installing the opendatacon in order to be able to run it. What am I missing?

**Comment by neilstephens** [comment]
Which version/branch are you building, and what OS are you using?

The release versions require manual installation of the init script, and
I've only tested it on debian and redhat. It relies on opendatacon being
installed at /opt/ODC

If you're on develop, the ODCdaemon script and init script has actually
been deprecated, because opendatacon will now run naively as a daemon (or
service under windows). And it also uses cpack to generate installers, so
it's up to the user to create an init script if required.

I'll delete the deprecated scripts and add a task to look into
automatically installing an init script (this would match the windows
option of installing the service).

Thanks for your feedback.
Neil.

**Comment by neilstephens** [comment]
By the way, I should have also said you can always run opendatacon from the command line.

**Comment by rrivera1849** [comment]
Thanks for your reply Neil!

I am now attempting to run a simple opendatacon example. I used the DMC example provided as a template, this is a much smaller test case with only one DNP3 Master and one DNP3 outstation.

[opendatacon](https://github.com/neilstephens/opendatacon/files/485746/opendatacon.txt)
[Connector](https://github.com/neilstephens/opendatacon/files/485747/Connector.txt)
[DNP3_Master](https://github.com/neilstephens/opendatacon/files/485748/DNP3_Master.txt)
[DNP3_Outstation](https://github.com/neilstephens/opendatacon/files/485749/DNP3_Outstation.txt)
[Log0](https://github.com/neilstephens/opendatacon/files/485750/Log0.txt)

When I run the opendatacon, this is the output I get:
`2016-09-21 11:13:00.70 - DEBUG - DNP3MasterPort - master: Setting point quality to COMM_LOST
2016-09-21 11:13:00.71 - DEBUG - outst_0.0.0.0:20001 - MonitorStateInit -> MonitorStateOpening
2016-09-21 11:13:18.872 - DEBUG - outst_0.0.0.0:20001 - MonitorStateOpening -> MonitorStateOpeningSuspending
2016-09-21 11:13:18.872 - WARN - outst_0.0.0.0:20001 - error: Operation aborted.
2016-09-21 11:13:18.872 - DEBUG - outst_0.0.0.0:20001 - Failed socket open, closing socket
2016-09-21 11:13:18.872 - DEBUG - outst_0.0.0.0:20001 - MonitorStateOpeningSuspending -> MonitorStateSuspended
2016-09-21 11:13:18.873 - DEBUG - outst_0.0.0.0:20001 - MonitorStateSuspended -> MonitorStateShutdown
2016-09-21 11:13:18.873 - DEBUG - mast_127.0.0.1:20000 - MonitorStateInit -> MonitorStateShutdown`

Is there anything wrong in my configuration files? 

It might be useful to know that when I run the DNP3Port_tests I get the following output:

`~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
DNP3Port_tests is a Catch v1.3.2 host application.
Run with -? for options

DNP3PortEndToEndTestSuite - ConstructEnableDisableDestroy
/home/tmauser/opendatacon/tests/DNP3Port_tests/TestDNP3PortEndToEnd.cpp:30

/home/tmauser/opendatacon/tests/DNP3Port_tests/TestDNP3PortEndToEnd.cpp:68: FAILED:
  REQUIRE( OPUT->GetStatus()["Result"].asString() == "Port enabled - link up (unreset)" )
with expansion:
  "Port enabled - link down"
  ==
  "Port enabled - link up (unreset)"

test cases: 3 | 2 passed | 1 failed
assertions: 8 | 7 passed | 1 failed
`

In the end, I would like to use the opendatacon to aggregate data from multiple DNP3 outstations, any advice that you could provide me to accomplish this would be extremely beneficial.

I thank you for your patience and responsiveness!

**Comment by rrivera1849** [comment]
I tried the netcat command, no errors were reported. I also ran the opendatacon as superuser. 

Here is a dump of my system-info:
[ls.pci](https://github.com/neilstephens/opendatacon/files/486319/ls.pci.txt)
[lsb.release](https://github.com/neilstephens/opendatacon/files/486322/lsb.release.txt)
[proc.cpuinfo](https://github.com/neilstephens/opendatacon/files/486320/proc.cpuinfo.txt)
[proc.meminfo](https://github.com/neilstephens/opendatacon/files/486321/proc.meminfo.txt)
[proc.version](https://github.com/neilstephens/opendatacon/files/486323/proc.version.txt)

I am running this on a Debian 64 bit VM with virtualbox.

**Comment by neilstephens** [comment]
I just fixed the end-to-end DNP3 port unit test. Please check out the latest develop commit and rebuild.

If the unit test fails for you now, it's safe to say it will be due to the same error you're getting on your example config.

Thanks for your system info. Can you also tell me your compiler version? Also, if you run cpack ('make package' in your cmake build dir) and send me the generated installer, I can try running your build, which could rule out some things.

And vice-versa - give this build a go: 
[opendatacon-0.3.4-Linux-x86_64-Release.tar.gz](https://github.com/neilstephens/opendatacon/files/486653/opendatacon-0.3.4-Linux-x86_64-Release.tar.gz)
 

Cheers,
Neil.

**Comment by rrivera1849** [comment]
I pulled the latest changes and the DNP3Port_tests are working perfectly. I then tried to run it with both my build and your build but it didn't work :(

**GCC Version:**
gcc (Debian 4.9.2-10) 4.9.2
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**G++ Version:**
g++ (Debian 4.9.2-10) 4.9.2
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**My build:**
[opendatacon-0.3.4-Linux-x86_64-.tar.gz](https://github.com/neilstephens/opendatacon/files/487725/opendatacon-0.3.4-Linux-x86_64-.tar.gz)

I will try to start a clean vm later today and see if your build works there. Thank you so much for your help Neal!

Rafael Rivera-Soto

**Comment by rrivera1849** [comment]
Alright, another status update. I did a clean reinstall of the Debian 64 bit VM and attempted to run the build once again but it doesn't seem to work :( 

When I query the status of the master or outstation using the command line plugin, this is what I get:
odc DataPorts> Status master
master:
{
   "Result" : "Port enabled - link down"
}

Perhaps I will try to install this locally on my MAC and see what happens.

**Comment by neilstephens** [comment]
I noticed you didn't have a build type set for your build (Release/Debug). This resulted in the opendnp3 libraries not being packaged.

So I set a default for that variable, and while I was at it, I tidied up all the cmake cache variables, so it should look simpler in a cmake gui now. I pushed it to the develop branch.

Now that we've eliminated the build itself being the cause of your problem, I'm going to close this issue. We can take further troubleshooting offline, and if it turns out to be a problem with opendatacon, we can open a new issue, or reopen this one.

Cheers,
Neil.
